loris/img.py

```diff
@@ -215,21 +215,21 @@ def __getitem__(self, image_request):
 		return fp
 
 	@staticmethod
-	def _link(to,frum):
+	def _link(to, frum):
 		link_dp = path.dirname(frum)
 		if not path.exists(link_dp):
 			makedirs(link_dp)
 		if path.lexists(frum): # shouldn't be the case, but helps debugging
 			unlink(frum)
-		symlink(to,frum)
-		logger.debug('Made symlink from %s to %s' % (to,frum))
+		symlink(to, frum)
+		logger.debug('Made symlink from %s to %s' % (frum, to))
 
-	def __setitem__(self, image_request, fp):
-		# The Image (fp) already exists, this simply makes a symlink in the
-		# cache to from the canonical syntax to the actual request path
+	def __setitem__(self, image_request, canonical_fp):
```

Comment:
Yes, I agree, this is the one place where my half-baked effort to implement the dict API breaks a little bit. The problem is that PIL and Kakadu need a path for the output image, and it doesn't make any sense to move the file (elsewhere to get it "in" to the cache)  once it's been made. The workaround is that the [webapp asks the cache for a path](https://github.com/loris-imageserver/loris/blob/development/loris/webapp.py#L588) based on the request up front, `__setitem__` is called later on when we know everything worked.

I'm more than happy to hear alternatives! Thanks for the review!